### PR TITLE
Feature/buttons disabled option

### LINKED ISSUE #582: Disabled option for buttons
It would be helpful if the StepOptionsButton had a disabled property. This property could also be a function to allow more complex logic.
```typescript
disabled: boolean | (() => boolean);
```

If the property is true, the button would have the `disabled` property to true.

### PR DESCRIPTION
**Pull request by VladimirAnaniev**
Closes #582  
**Screenshot of a disabled button**  
Screenshot from 2019-10-02 22-22-34

### CODE REVIEW

**Comment by rwwagner90** [comment]
@VladimirAnaniev thanks for the PR! If you haven't already, we are running a contest for giving away some Ember Temperature Control Mugs for Hacktoberfest, if you would like to enter. Opening a PR for a Hacktoberfest issue gives you 100 entries! https://twitter.com/shipshapecode/status/1179190723666759681

File: src/types/step.d.ts [hunk 1]
```diff
@@ -216,6 +216,12 @@ declare namespace Step {
      * The HTML text of the button
      */
     text?: string;
+
+    /**
+     * Whether the button should be disabled
+     * When the value is `true` the button will be disabled
+     */
+    disabled?: boolean;
```
Comment (rwwagner90) [inline src/types/step.d.ts]: Didn't we want to allow passing functions for disabled as well?

Comment (VladimirAnaniev) [reply src/types/step.d.ts]: I thought about it but was not sure when we should re-evaluate the function.  
When do you think is suitable? On every show event? On an interval?

Comment (rwwagner90) [reply src/types/step.d.ts]: I would think we would do the same thing we do with `text`.

File: test/cypress/integration/test.acceptance.js [hunk 2]
```diff
@@ -134,6 +134,56 @@ describe('Shepherd Acceptance Tests', () => {
...
+              {
+                classes: 'enabled-button',
+                text: 'enabled'
+              }
+            ]
+          }
+        ]);
+        tour.start();
+        tour.show('test-step');
+        cy.get('.shepherd-button.disabled-button')
```
Comment (rwwagner90) [inline test/cypress/integration/test.acceptance.js]: Would we be able to move these tests to a unit test of the shepherd-button component? They are fairly simple, and unit tests will add to coverage.

Comment (VladimirAnaniev) [reply test/cypress/integration/test.acceptance.js]: Yes, this makes sense, I have moved them :)

**Comment by rwwagner90** [comment]
Thanks so much for the PR @VladimirAnaniev! Really appreciate your help with this feature 😃

**Pull request merged**